internal/dinosql/config.go

```diff
@@ -10,11 +10,27 @@ import (
 	"github.com/kyleconroy/sqlc/internal/pg"
 )
 
+const errMessageNoVersion = `The configuration file must have a version number.
+Set the version to 1 at the top of sqlc.json:
+
+{
+  "version": "1"
+  ...
+}
+`
+
+const errMessageUnknownVersion = `The configuration file has an invalid version number.
+The only supported version is "1".
+`
+
+const errMessageNoPackages = `No packages are configured`
+
 type GenerateSettings struct {
-	Version   string            `json:"version"`
-	Packages  []PackageSettings `json:"packages"`
-	Overrides []Override        `json:"overrides,omitempty"`
-	Rename    map[string]string `json:"rename,omitempty"`
+	Version    string            `json:"version"`
+	Packages   []PackageSettings `json:"packages"`
+	Overrides  []Override        `json:"overrides,omitempty"`
+	Rename     map[string]string `json:"rename,omitempty"`
+	PackageMap map[string]PackageSettings
```

Comment:
I'm not sure I understand the use case for `PackageMap`. Is it just for convenience?

Reply:
Answered my own question: The package map means we can pass around the settings struct by itself, instead of a global and package settings pair

Reply:
Exactly, my original thinking was that it would make lookups by package name easier, but I suppose the argument could be made that it made things a bit more indirect and error prone than they should be.